lib/cacheable/controller.rb

```diff
@@ -36,14 +36,25 @@ def response_cache(key_data=nil, version_data=nil, &block)
         return yield
       end
 
-      handler = Cacheable::ResponseCacheHandler.new(self) do |h|
-        h.key_data       = key_data       || cache_key_data
-        h.version_data   = version_data   || cache_version_data
-        h.block          = block
-        h.cache_store    = Cacheable.cache_store
-      end
+      handler = Cacheable::ResponseCacheHandler.new(
+        self,
+        key_data:              key_data || cache_key_data,
+        version_data:          version_data || cache_version_data,
+        env:                   request.env,
+        cache_age_tolerance:   cache_age_tolerance_in_seconds,
+        serve_unversioned:     serve_unversioned_cacheable_entry?,
+        force_refill_cache:    force_refill_cache?,
+        headers:               response.headers,
+        &block
+      )
 
       handler.run!
```

Comment:
I’m beginning to wonder if this should just be a class method that accepts all the args. Maybe this means we won’t need instance_exec, and thus not need to pass in the controller instance.

Reply:
I managed to remove the instance_eval by moving the `render` and `head` calls to `Cacheable::Controller` 🎉 

I can add a class method as a short hand but I think the instance is useful